src/Enemy/EnemyCap.cpp

```diff
@@ -0,0 +1,186 @@
+void EnemyCap::initPartsFixFile(al::LiveActor* actor, const al::ActorInitInfo& initInfo, const char* archiveName, const char* suffix) {
+    mCap = actor;
+    mCapBaseMtx = mCap->getBaseMtx();
+    al::initChildActorWithArchiveNameNoPlacementInfo(this, initInfo, archiveName, 0);
+    al::initNerve(this, &NrvEnemyCap.Wait, 1);
+    mStateBlowDown = new al::EnemyStateBlowDown(this, fallbackStateBlowDownParam, "吹き飛び状態");
+    al::initNerveState(this, mStateBlowDown, &NrvEnemyCap.BlowDown, "吹き飛び");
+
+    al::StringTmp<128> test("");
+
+    al::createFileNameBySuffix(&test, "InitPartsFixInfo", suffix);
+    if (al::isExistModelResourceYaml(mCap, test.cstr(), 0)) {
+        al::ByamlIter resourceYaml = al::getModelResourceYaml(mCap, test.cstr(), 0);
+        const char* jointName = 0;
+        resourceYaml.tryGetStringByKey(&jointName, "JointName");
+        if (jointName)
+            mCapBaseMtx = al::getJointMtxPtr(mCap, jointName);
+        al::tryGetByamlV3f(&mLocalTrans, resourceYaml, "LocalTrans");
+        al::tryGetByamlV3f(&mLocalRotate, resourceYaml, "LocalRotate");
+        al::tryGetByamlV3f(&mLocalScale, resourceYaml, "LocalScale");
+        if (!al::isNearZero(mLocalTrans, 0.001f) || !al::isNearZero(mLocalRotate, 0.001f))
+            mIsAtOrigin = true;
+        mUseFollowMtxScale = al::tryGetByamlKeyBoolOrFalse(resourceYaml, "UseFollowMtxScale");
+        mUseLocalScale = al::tryGetByamlKeyBoolOrFalse(resourceYaml, "UseLocalScale");
+    }
+    makeActorAlive();
+}
+
+void EnemyCap::makeActorAlive() {
+    al::invalidateClipping(this);
+    al::offCollide(this);
+    al::setVelocityZero(this);
+    updatePose();
+    al::LiveActor::makeActorAlive();
+    this->mCapVisible = 0;
+    al::setNerve(this, &NrvEnemyCap.Wait);
+}
+
+void EnemyCap::calcAnim() {
+    if (al::isNerve(this, &NrvEnemyCap.Wait)) {
+        updatePose();
+    }
+    al::LiveActor::calcAnim();
+}
+
+void EnemyCap::exeWait() {
+    if (!syncHostVisible())
+        return;
+    al::setModelAlphaMask(this, al::getModelAlphaMask(mCap));
+}
+
+bool EnemyCap::syncHostVisible() {
+    return al::updateSyncHostVisible(&mCapVisible, this, mCap, field_142);
+}
+
+void EnemyCap::exeBlowDown() {
+    if (al::isFirstStep(this)) {
+        al::onCollide(this);
+        al::offSyncClippingSubActor(mCap, this);
+        al::offSyncAlphaMaskSubActor(mCap, this);
+    }
+    al::rotateQuatXDirDegree(this, -15.0f);
+    if (!al::updateNerveState(this))
+        return;
+    al::offCollide(this);
+    al::startHitReaction(this, "消滅");
+    al::onSyncClippingSubActor(mCap, this);
+    al::onSyncAlphaMaskSubActor(mCap, this);
+    kill();
+}
+
+void EnemyCap::startBlowDown(const al::HitSensor* source) {
+    if (isBlowDown())
+        return;
+    mStateBlowDown->start(source);
+    al::setNerve(this, &NrvEnemyCap.BlowDown);
+}
+
+void EnemyCap::startBlowDown() {
+    if (isBlowDown())
+        return;
+    mStateBlowDown->start(mCap);
+    al::setNerve(this, &NrvEnemyCap.BlowDown);
```

Comment:
```suggestion
    startBlowDown(mCap);
```

Reply:
This doesn't work because `mCap` is an `al::LiveActor`, and EnemyCap does not have a function that accepts that.

Reply:
Oh, alright, was thinking about the function taking `HitSensor` without looking further into it.